closed/adds/jdk/src/windows/native/jdk/crypto/jniprovider/NativeCrypto_md.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2019, 2019 All Rights Reserved
+ * ===========================================================================
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * IBM designates this particular file as subject to the "Classpath" exception
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+
+#include <windows.h>
+
+#include "NativeCrypto_md.h"
+
+/* Load the crypto library (return NULL on error) */
+void * load_crypto_library() {
+    void * result = NULL;
+    const char *libname;
+    const char *oldname = "libeay32.dll";
+
+#ifdef WIN32
```

Comment:
Are we sure that `WIN32` is *not* defined for 64-bit builds? There's probably a better (and clearer) way to distinguish between 32-bit and 64-bit builds.

Reply:
There is some JNI code which uses IsWow64BitProcess to work out if it is running in a 64-bit process. Even that ends us using #ifdef _WIN64 to insert code to pick up the correct message to throw. I would assume _WIN64 is not set for 32 bit builds - See https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2017

Reply:
Using _WIN64, which will require some testing.